apps/bfd-db-migrator/src/main/java/gov/cms/bfd/migrator/app/HibernateValidator.java

```diff
@@ -0,0 +1,154 @@
+   * @param modelPackagesToScan the model packages to scan
+   */
+  public HibernateValidator(HikariDataSource dataSource, List<String> modelPackagesToScan) {
+    this.dataSource = dataSource;
+    this.modelPackagesToScan.addAll(modelPackagesToScan);
+    this.schemaValidator = new SchemaValidator();
+    this.hibernateConfiguration = new Configuration();
+  }
+
+  /**
+   * Sets the schema validator, primarily for testing.
+   *
+   * @param schemaValidator the schema validator
+   */
+  public void setSchemaValidator(SchemaValidator schemaValidator) {
+    this.schemaValidator = schemaValidator;
+  }
+
+  /**
+   * Sets the hibernate configuration, primarily for testing.
+   *
+   * @param configuration the configuration
+   */
+  public void setHibernateConfiguration(Configuration configuration) {
+    this.hibernateConfiguration = configuration;
+  }
+
+  /**
+   * Runs hibernate validation and reports if it succeeded.
+   *
+   * @return {@code true} if the validation succeeded
+   */
+  public boolean runHibernateValidation() {
+    try {
+      // Add the models to scan for (used in validation)
+      Set<Class<?>> scannedClasses = new HashSet<>();
+
+      for (String packagePath : modelPackagesToScan) {
+        scannedClasses.addAll(getEntityClassesFromPackage(packagePath));
+      }
+
+      if (scannedClasses.isEmpty()) {
+        LOGGER.error("Found no classes to validate.");
+        return false;
+      }
+      LOGGER.debug("Added {} classes to be validated.", scannedClasses.size());
+
+      SessionFactory sessionFactory = createHibernateSessionFactory(dataSource, scannedClasses);
+      // Validate the metadata
+      StandardServiceRegistry registry =
+          sessionFactory.getSessionFactoryOptions().getServiceRegistry();
+      MetadataSources sources = new MetadataSources(registry);
+      Metadata metadata = sources.buildMetadata(registry);
+      // This will throw an exception if validation fails
+      schemaValidator.validate(metadata);
+    } catch (HibernateException hx) {
+      LOGGER.error("Hibernate validation failed due to: ", hx);
+      return false;
+    } catch (Exception ex) {
+      LOGGER.error("Hibernate validation failed due to unexpected exception: ", ex);
+      return false;
+    }
+    return true;
+  }
+
+  /**
+   * Creates a Hibernate session factory, needed for obtaining the metadata used in validation.
+   *
+   * @param dataSource the data source which contains the database connection
+   * @param classesToValidate the classes to validate
+   * @return the session factory
+   */
+  private SessionFactory createHibernateSessionFactory(
+      DataSource dataSource, Set<Class<?>> classesToValidate) {
+
+    for (Class<?> clazz : classesToValidate) {
+      hibernateConfiguration.addAnnotatedClass(clazz);
+    }
+
+    // Set hibernate to validate the models on startup
+    hibernateConfiguration.setProperty(AvailableSettings.HBM2DDL_AUTO, "validate");
```

Comment:
do we still need this?

Reply:
Yes, apparently; I attempted to remove it and no validation ran.

Reply:
Chceked and we indeed still need this line; it will default to NONE if not supplied, which does not run the validation.

Reply:
thanks ; a comment here recapping this discussion might save some time in the future